### PR TITLE
[Merged by Bors] - feat: `mapsTo`, `surjOn` and `injOn` are preserved by `Filter.map`

### PR DESCRIPTION
**Pull request by ADedecker**
And use it to simplify the proof of [completeSpace_iff_isComplete_range](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/UniformSpace/UniformEmbedding.html#completeSpace_iff_isComplete_range).

I also add the consequence of the previous result that any uniform embedding from a complete space to a hausdorff space is a closed embedding (I agree this may be a bit unrelated, but I was in the area).

Notes about the filter results:
- the left/right inverse results are not used, but I think they can be helpful. I'm okay with removing them.
- the new API doesn't match very well the existing one, but after way more time that I'm willing to admit I came to the conclusion that it doesn't make sense to make it better without reorganizing and splitting the whole file.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/UniformSpace/CompleteSeparated.lean [hunk 1]
```diff
@@ -5,6 +5,7 @@ Authors: Sébastien Gouëzel
 -/
 import Mathlib.Topology.UniformSpace.Cauchy
 import Mathlib.Topology.UniformSpace.Separation
+import Mathlib.Topology.UniformSpace.UniformEmbedding
 import Mathlib.Topology.DenseEmbedding
```
Comment (fpvandoorn) [inline Mathlib/Topology/UniformSpace/CompleteSeparated.lean:9]:
```suggestion
import Mathlib.Topology.UniformSpace.UniformEmbedding
```

File: Mathlib/Order/Filter/Basic.lean [hunk 2]
```diff
@@ -3296,6 +3327,30 @@ theorem Filter.EventuallyEq.comp_tendsto {f' : α → β} (H : f =ᶠ[l] f') {g
...
+
+-- TODO(Anatole): unify with the global case
+
+theorem Filter.map_surjOn_Iic_of_le_map {m : α → β} (hm : G ≤ map m F) :
+    SurjOn (map m) (Iic F) (Iic G) := by
+  have : RightInvOn (F ⊓ comap m ·) (map m) (Iic G) :=
+    fun H (hHG : H ≤ G) ↦ by simpa [Filter.push_pull] using hHG.trans hm
+  exact this.surjOn fun H _ ↦ mem_Iic.mpr inf_le_left
+
+theorem _root_.Set.SurjOn.filter_map {m : α → β} (hm : SurjOn m s t) :
```
Comment (fpvandoorn) [inline Mathlib/Order/Filter/Basic.lean:3346]:
```suggestion
theorem _root_.Set.SurjOn.filter_map_Iic {m : α → β} (hm : SurjOn m s t) :
```
The name should probably mention `Iic` (also for other lemmas).

**Review by fpvandoorn — COMMENTED** [review]
Do you think any of the new lemmas to `Filter.Basic` could be usefully upgraded to `iff`-lemmas?

bors d+

**Comment by ADedecker** [comment]
I added the `iff` version of the `mapsTo`, `injOn` and `surjOn` lemmas, but not of the `[Left, Right]Inverse` lemmas because I don't think they're that useful, and they're pretty easy to add if needed.

**Pull request closed without merging**